### PR TITLE
Let `:nessie-versioned-spi` use OpenTelemetry

### LINKED ISSUE #5608: Use OpenTelemetry in `:nessie-versioned-spi`
Migrate the existing code from OpenTracing & Micrometer to OpenTelemetry.
Once that's done, the shim-code in `:nessie-quarkus` can be removed, and the dependency references to micrometer + opentracing removed from `libs.gradle.toml`.

### PR DESCRIPTION
**Pull request by snazy**
Migrate from OpenTracing to OpenTelemetry.

Updates some build files to refer to "one of the right opentelemetry BOMs" (there are 4 or so, but all interchangeable).

Fixes #5608

### CODE REVIEW

File: versioned/spi/src/main/java/org/projectnessie/versioned/TracingVersionStore.java [hunk 1]
```diff
@@ -138,11 +147,12 @@ public MergeResult<Commit> transplant(
     return TracingVersionStore
         .<MergeResult<Commit>, ReferenceNotFoundException, ReferenceConflictException>
             callWithTwoExceptions(
```
Comment (dimas-b) [inline versioned/spi/src/main/java/org/projectnessie/versioned/TracingVersionStore.java:149]: I wonder whether we could use simple delegation code here with method-level `@WithSpan` annotations and instrument the bytecode with OTel at build time... WDYT?

Span attributes can be added via `Span.current().setAttribute(...)`

File: versioned/spi/src/test/java/org/projectnessie/versioned/TestTracingVersionStore.java [hunk 2]
```diff
@@ -259,89 +263,125 @@ void versionStoreInvocation(
...
+      //                expectedLogs, tracer.getActiveSpan().getLogs(), "expected logs don't
+      // match"),
+      //        () ->
+      //            assertEquals(
+      //                new HashMap<>(expectedTags),
+      //                tracer.getActiveSpan().getTags(),
+      //                "expected tags don't match"),
+      //        () -> assertTrue(tracer.isParentSet(), "Span-parent not set"),
+      //        () -> assertTrue(tracer.isClosed(), "Scope not closed"),
+      //        () -> assertTrue(tracer.getActiveSpan().finished(), "Span did not finish"));
```
Comment (dimas-b) [inline versioned/spi/src/test/java/org/projectnessie/versioned/TestTracingVersionStore.java:383]: Do we want to assert this?

**Review by dimas-b — COMMENTED** [review]
LGTM - the `@WithSpan` comment is optional :)

File: versioned/spi/src/main/java/org/projectnessie/versioned/TracingVersionStore.java (hunk 1 above)
Comment (snazy) [reply versioned/spi/src/main/java/org/projectnessie/versioned/TracingVersionStore.java:149]: In Quarkus, probably better, yea. Let's get that done for Quarkus.

Comment (dimas-b) [reply versioned/spi/src/main/java/org/projectnessie/versioned/TracingVersionStore.java:149]: AFAIK, instrumentation can be done in a generic way with Quarkus providing runtime OTel impl.

**Review by dimas-b — APPROVED** [review]

File: versioned/spi/src/main/java/org/projectnessie/versioned/TracingVersionStore.java (hunk 1 above)
Comment (snazy) [reply versioned/spi/src/main/java/org/projectnessie/versioned/TracingVersionStore.java:149]: Yep, there's a way to do that.

**Pull request merged**